### PR TITLE
Sys.Reboot: support abort()ing

### PR DESCRIPTION
**Pull request by QRPp**
Add a boolean “abort” RPC argument that, if set, causes an abort() instead of a graceful reboot.

Good for shedding coredumps with a minimal state disruption.

### CODE REVIEW

File: src/mgos_rpc.c [hunk 1]
```diff
@@ -154,17 +156,25 @@ static void mgos_rpc_http_handler(struct mg_connection *nc, int ev,
...
   int delay_ms = 100;
-  json_scanf(args.p, args.len, ri->args_fmt, &delay_ms);
+  json_scanf(args.p, args.len, ri->args_fmt, &abort, &delay_ms);
   if (delay_ms < 0) {
     mg_rpc_send_errorf(ri, 400, "invalid delay value");
     ri = NULL;
     return;
   }
-  mgos_system_restart_after(delay_ms);
+  if (!abort)
```
Comment (rojer) [inline src/mgos_rpc.c]: please add braces

File: src/mgos_rpc.c [hunk 2]
```diff
@@ -154,17 +156,25 @@ static void mgos_rpc_http_handler(struct mg_connection *nc, int ev,
 #endif /* defined(MGOS_HAVE_HTTP_SERVER) && MGOS_ENABLE_RPC_CHANNEL_HTTP */
 
 #if MGOS_ENABLE_SYS_SERVICE
+static void mgos_sys_reboot_abort_tmr(void *opaque) {
+  abort();
```
Comment (rojer) [inline src/mgos_rpc.c:161]: add a log info line just before announcing that abort is intentional.

File: src/mgos_rpc.c (hunk 1 above)
Comment (QRPp) [reply src/mgos_rpc.c]: Done.

File: src/mgos_rpc.c (hunk 2 above)
Comment (QRPp) [reply src/mgos_rpc.c:161]: Done; however not here but in the RPC handler: that way the logging is in the same manner as that for rebooting normally.

Comment (rojer) [reply src/mgos_rpc.c:161]: i think it's nice to also have a log line just before the abort itself, like we have for reboot here - https://github.com/cesanta/mongoose-os/blob/master/src/mgos_system.c#L43-L44

**Comment by rojer** [comment]
merged, thanks!

**Pull request merged**